File: especialistaapi_backend/src/main/java/cl/giancarlo/especialistaapi/models/entities/Especialista.java
package cl.giancarlo.especialistaapi.models.entities;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "especialistas")
public class Especialista {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(nullable = false)
    private String nombre;

    @Column(nullable = false)
    private String apellido;

    @Column(nullable = false)
    private String especialidad;

    @Column(nullable = false, unique = true)
    private String email;

    @Column(nullable = false)
    private String telefono;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Modalidad modalidad;

    @Column(nullable = false)
    private int tarifaHora;

}

File: AppAndroid/app/src/main/java/cl/giancarlo/especialista/models/Especialista.kt
package cl.giancarlo.especialista.models

data class Especialista(
    val id: Int,
    val nombre: String,
    val apellido: String,
    val especialidad: String,
    val email: String,
    val telefono: String,
    val modalidad: Modalidad,
    val tarifaHora: Int
)